test/python/providers/test_fake_backends.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=missing-class-docstring,missing-function-docstring
+# pylint: disable=missing-module-docstring
+
+import operator
+
+from test import combine
+from ddt import ddt
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.execute import execute
+from qiskit.test.base import QiskitTestCase
+from qiskit.test.mock import backends
+from qiskit.test.mock.fake_backend import HAS_AER
+
+
+FAKE_BACKENDS = [
+    x for x in dir(backends) if x.startswith('Fake')]
+
+
+@ddt
+class TestFakeBackends(QiskitTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.circuit = QuantumCircuit(2)
+        cls.circuit.h(0)
+        cls.circuit.h(1)
+        cls.circuit.h(0)
+        cls.circuit.h(1)
+        cls.circuit.x(0)
+        cls.circuit.x(1)
+        cls.circuit.measure_all()
+
+    @combine(fake_backends=FAKE_BACKENDS,
+             optimization_level=[0, 1, 2, 3])
+    def test_circuit_on_fake_backend(self, fake_backends, optimization_level):
+        backend = getattr(backends, fake_backends)()
+        if not HAS_AER and backend.configuration().n_qubits > 20:
```

Comment:
It's a bit unfortunate that we need to use (and install) Aer for these tests (though I understand why its necessary). Maybe it makes sense to update the BasicAer n_qubits check to be number of used qubits.

Reply:
Yeah, I wasn't happy with adding aer to the testing. But to do this well it was needed.

The issue with adjusting basicaer to avoid this is that it's looking at the transpiled qobj to determine the number of qubits required: https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/providers/basicaer/qasm_simulator.py#L622. With the fake backends (and specifically in this case for FakeRochester), the physical circuit will have an ancilla with no operations set for all the unused qubits since it's pulling that from the fake backend. To workaround this we'd have to disassemble the qobj and determine that 51 qubits are idle. It's not too much work, but definitely outside the scope of this PR.